forc/src/ops/forc_build.rs

```diff
@@ -160,6 +161,15 @@ pub fn build(command: BuildCommand) -> Result<pkg::Compiled> {
 
     info!("  Bytecode size is {} bytes.", compiled.bytecode.len());
 
+    if let TreeType::Script | TreeType::Predicate = compiled.tree_type {
+        // hash the bytecode for scripts/predicates and store it in a file in the output directory
+        let bytecode_hash = format!("0x{}", fuel_crypto::Hasher::hash(&compiled.bytecode));
+        let hash_file_name = format!("{}-bin-hash", &manifest.project.name);
+        let hash_path = output_dir.join(hash_file_name);
+        fs::write(hash_path, &bytecode_hash)?;
```

Comment:
Can you file an issue to explore how best to handle this? It's not necessarily the case that we want this file emitting by default, as usually users won't actually need to script or predicate hash manually.

Reply:
I think this should be fine - it's a tiny file and will be enclosed in the `out/<profile>/` directory, so hopefully not too intrusive. I'm happy to leave it in for this PR at least.